packer/builder/azure/arm/step_delete_resource_group.go

```diff
@@ -30,12 +31,34 @@ func NewStepDeleteResourceGroup(client *AzureClient, ui packer.Ui) *StepDeleteRe
 }
 
 func (s *StepDeleteResourceGroup) deleteResourceGroup(resourceGroupName string) error {
-	res, err := s.client.GroupsClient.Delete(resourceGroupName)
+	_, err := s.client.GroupsClient.Delete(resourceGroupName)
 	if err != nil {
 		return err
 	}
 
-	s.client.GroupsClient.PollAsNeeded(res.Response)
+	return s.waitForResourceGroupToBeDeleted(resourceGroupName)
+}
+
+// The API does not correctly poll for completion of the Delete command for the groups client, so here's a workaround.
+//
+// There are two polling variations that I am aware of.
+//  1. API returns a location header and HTTP status code of 202.  Client must poll the location until complete.
+//  2. API returns an azure async operation header.
+func (s *StepDeleteResourceGroup) waitForResourceGroupToBeDeleted(resourceGroupName string) error {
+	deadline := time.Now().Add(15 * time.Minute)
+	for {
+		if time.Now().After(deadline) {
+			return fmt.Errorf("Exceeded poll duration while waiting for resource group %s to be deleted", resourceGroupName)
+		}
+
+		_, err := s.client.GroupsClient.Get(resourceGroupName)
+		if err != nil {
+			break
```

Comment:
wouldn't you want to check that this is a specific error, e.g. not a 500 or something?